bitmark-references/bitmark-node-software/bitmark-blockain.md

```diff
@@ -0,0 +1,125 @@
+# Blockchain
+
+## Block
+
+Block is a bunch of transactions that have been added to blockchain. A bitmark block contains data as follows:
+
+| size (bytes) | data                                                           |
+|--------------|----------------------------------------------------------------|
+|            2 | block version number                                           |
+|            2 | transaction count inside the block                             |
+|            8 | block number (block height)                                    |
+|           32 | argon2 hash of previous block header                           |
+|           32 | sha3 hash of all transactions in the block (merkle root)       |
+|            8 | timestamp of block generation time, since 1970-01-01T00:00 UTC |
+|            8 | current difficulty in compact mode                             |
+|            8 | number to make hash meets difficulty                           |
+|   ... to end | transactions                                                   |
+
+Different types of transaction includes different lenegh and content inside.
+
+### Block Verification
+
+When a block is received, following checks will be made:
+
+1. Length of all parts inside a block should match
+1. Incoming block version number should be equal or largen than current block version number
+1. Incoming block height is 1 ahead of current block height
+1. Incoming block contans at least 1 transaction, at most 9999 transactions inside
+1. Incoming block generation time no further than 5 minutes after received time
+1. Incoming block generation time can not be larger then 10 minutes compares to current block generation time
+1. Incoming block generation time within specific range
+1. Incoming block difficulty is same as current difficulty
+1. Incoming block hash meets difficulty criteria
+1. Incoming block's previous block hash same as current block hash
+1. Incoming transaction fits each type's format and length
+1. Incoming transaction owner ownership is valid
+1. Incoming transaction payment is valid
+1. Incoming block's sha3 of all transactions inside is valid
+
```

Comment:
>Incoming block generation time no further than 5 minutes after received time

This statement is hard to understand.

Reply:
Are these criteria applied for only for the new block? I am confused because I think the block synchronization also refers to the process when a new node joins the network, it will need to synchronize the blocks from the genesis one. If I understand correctly, the following criteria seem to not correct
```1. Incoming block generation time no further than 5 minutes after received time
1. Incoming block difficulty is same as current difficulty```

Reply:
1. change the description, see if this makes easier to read

2. all blocks needs to pass these verifications. I am not sure the contradiction above two rules?

Reply:
I actually don't get the meaning of the following criterion
>Incoming block generation time no further than 5 minutes after received time

About the second one:
> Incoming block difficulty is same as current difficulty

If I am a node which has just joined the network for the first time and synchronizing the blocks from the beginning. I am looking for block #1000 while the blockchain height is now 2000. So the difficulty of the block #1000 might be different from the current difficulty, right?

Reply:
For the first one, time on network is not consistent, so 5 minutes is a buffer in case other node's time is different from the receiving node.

For second one, the node should verify difficulty based on the incoming block number, in the example above, the node needs to verify block number 1000 hash meets block number 1000 difficulty. When start to sync from block number 1000, the node will calculate difficulty based on previous blocks, and make sure incoming new block meets local calculated difficulty

Reply:
I would agree with @vienlt — blocks are usually validated. The terms "validate" and "verify" are often used interchangeably in the blockchain world, but that seems to be the most common work for checking if  block is valid.

Reply:
I am not sure the meaning, is there wording I can change for better understanding?

Reply:
I suggest "Block Validation" instead of "Block Verification" and say that blocks are "validated" instead of "verified"

Reply:
changed